controller/src/main/java/io/pravega/controller/server/eventProcessor/requesthandlers/DeleteStreamTask.java

```diff
@@ -53,7 +53,11 @@ public DeleteStreamTask(final StreamMetadataTasks streamMetadataTasks,
         String scope = request.getScope();
         String stream = request.getStream();
         long requestId = request.getRequestId();
-        return streamMetadataStore.isSealed(scope, stream, context, executor)
+
+        return streamMetadataStore.getCreationTime(scope, stream, context, executor)
+            .thenAccept(creationTime -> Preconditions.checkArgument(request.getCreationTime() == 0 ||
```

Comment:
Using creation time is not bad, but perhaps we should consider having a more robust way, like a UUID to uniquely identify (whp) a stream. If you agree, let's create an issue before I approve it.

Reply:
I agree. I will create an issue for the same.

Reply:
Can you post the issue number here, pls?